tests/test_micrograph.py

```diff
@@ -0,0 +1,122 @@
+    )
+)
+def micrograph_fixture(sim_fixture, request):
+    """
+    Construct a MicrographSource
+    """
+    (
+        particles_per_micrograph,
+        micrograph_count,
+        micrograph_size,
+        boundary,
+    ) = request.param
+    return MicrographSource(
+        simulation=sim_fixture,
+        interparticle_distance=0,
+        particles_per_micrograph=particles_per_micrograph,
+        micrograph_count=micrograph_count,
+        micrograph_size=micrograph_size,
+        boundary=boundary,
+    )
+
+
+# Test the MicrographSource class takes and stores correct values.
+def test_micrograph_source_has_correct_values(sim_fixture, micrograph_fixture):
+    s = sim_fixture
+    m = micrograph_fixture
+    assert s.L == m.particle_box_size
+    assert s == m.simulation
+    assert m.particles_per_micrograph * m.micrograph_count <= s.n
+    assert len(m) == m.micrograph_count
+
+
+# Test that MicrographSource raises error when simulation argument is not a Simulation
+@pytest.mark.parametrize("fake_simulation", ("Simulation", 51378))
+def test_micrograph_raises_error_simulation(fake_simulation):
+    with pytest.raises(Exception) as e_info:
+        _ = MicrographSource(
+            fake_simulation,
+            micrograph_size=100,
+            particles_per_micrograph=20,
+            interparticle_distance=10,
+        )
+    assert str(e_info.value) == "Simulation should be of type Simulation"
+
+
+# Test the MicrographSource class raises errors when the image size is larger than micrograph size
+def test_micrograph_raises_error_image_size(sim_fixture):
+    with pytest.raises(Exception) as e_info:
+        s = sim_fixture
+        _ = MicrographSource(
+            s,
+            micrograph_size=s.L - 1,
+            particles_per_micrograph=10,
+            interparticle_distance=0,
+        )
+    assert (
+        str(e_info.value)
+        == "The micrograph size must be larger or equal to the simulation's image size."
+    )
+
+
+# Test the MicrographSource class raises errors when the interparticle_distance is too large
+def test_micrograph_raises_error_collisions(sim_fixture):
+    with pytest.raises(Exception) as e_info:
+        s = sim_fixture
+        _ = MicrographSource(
+            s,
+            micrograph_size=s.L * 10,
+            particles_per_microgaph=20,
+            interparticle_distance=10,
+        )
+        assert (
+            str(e_info.value)
+            == "The particle count times micrograph count must be less than or equal to the number o\
+f images generated by the simulation (particles_per_micrograph * micrograph_count <= simulation.n)"
+        )
+
+
+# Test that the Micrograph's centers are forming at generated points.
+def test_micrograph_centers_match(micrograph_fixture):
+    m = micrograph_fixture
+    centers = np.reshape(m.centers, (m.total_particle_count, 2))
+    for i, center in enumerate(centers):
+        if (
```

Comment:
fwiw, I think technically this test can fail, but in practice I really doubt it would often enough to care about doing something more difficult.